core/block/editor/file/file.go

```diff
@@ -476,6 +487,24 @@ func (dp *dropFilesProcess) Start(rootId, targetId string, pos model.BlockPositi
 			return
 		}
 		err = cache.Do(dp.picker, smartBlockIds[idx], func(sb File) error {
+			smartBlock, ok := sb.(smartblock.SmartBlock)
+			if ok {
+				if layout, ok := smartBlock.Layout(); ok && layout == model.ObjectType_collection {
+					for _, entry := range flatEntries[idx] {
+						if entry.isDir {
+							smartBlockIds = append(smartBlockIds, rootId)
```

Comment:
Why are we appending rootId? can we do it multiple times if we do it in a loop?

Reply:
If we do it only once we should structure the code in a way that it would be evident that this happens only once

Reply:
I would also extract this in a separate func